modules/noise/noise_texture_3d.cpp

```diff
@@ -142,6 +142,8 @@ TypedArray<Image> NoiseTexture3D::_generate_texture() {
 		return TypedArray<Image>();
 	}
 
+	ERR_FAIL_COND_V_MSG((int64_t)width * height * depth > std::numeric_limits<int>::max() / 8, TypedArray<Image>(), "The NoiseTexture3D is too big, consider lowering its width, height or depth.");
```

Comment:
I believe @emre0altan is referring to this (also parenthesis to be a bit clearer):
```suggestion
	ERR_FAIL_COND_V_MSG((int64_t)(width * height * depth) > Image::MAX_WIDTH, TypedArray<Image>(), "The NoiseTexture3D is too big, consider lowering its width, height or depth.");
```

Reply:
I think it should be `Image::MAX_PIXELS` instead, it would be the total allocation of the image.

Reply:
> And I think @jsjtxietian wanted to cast first so that it won't overflow.

Yes that's the reason why I wrote it this way. I will update my code now.